### PR TITLE
GUI fixes

### PR DESCRIPTION
**Pull request by AaronM04**
Fixes:
* fix intro centering
* fix video options alignment
* change some datatypes

Merge checklist:
- [x] **fix the issue with offset drawing when the gray area outside the grid is visible**
- [x] clean up `GridView` and `Viewport` in viewport.rs
- [x] clean up `VideoSettings` and `DisplayModeManager` in video.rs
- [x] ensure that if `fullscreen = true` in config on startup, that the logic is not reversed and it actually starts in fullscreen mode.
- [x] fix `draw_universe` TODO
- [x] use `f32` for window coordinates to match ggez
- [x] center intro even when fullscreen
- [x] center the menu
- [x] no warnings

### CODE REVIEW

File: conwayste/src/client.rs [hunk 1]
```diff
@@ -839,6 +846,7 @@ impl MainState {
 
     fn draw_universe(&mut self, ctx: &mut Context) -> GameResult<()> {
 
+        // TODO: don't do this every frame
```
[the remark this answers is not in the document]
Comment (Manghi) [reply conwayste/src/client.rs]: Resolved in fb81b09

File: conwayste/src/client.rs [hunk 2]
```diff
@@ -522,6 +518,7 @@ impl MainState {
             video_settings:      vs,
             config:              config,
             viewport:            viewport,
+            intro_viewport:      intro_viewport,
```
[the remark this answers is not in the document]
Comment (Manghi) [reply conwayste/src/client.rs]: Yeah I agree it's too early to optimize, and that the server should hold a lot of any optimization focus. Pinning this for another time.

**Comment by AaronM04** [comment]
@Manghi All the checkboxes are checked! This is ready for you to review. If it looks good, please merge.

**Comment by Manghi** [comment]
Looks great! Two questions:

1. Why do we not save the resolution to config when full screen?
2. The intro isn't centered on my system when fullscreen. Is that expected?

I ran this at the end of the `Intro` state when it transitions to `Run`. (I do agree we need to consolidate the billion different ways we store resolution.)

```Rust
println!("Viewport: {:?}", self.viewport.get_rect());
println!("Video Settings: {:?}", self.video_settings.get_resolution());
println!("Config: {:?}", self.config.get_resolution());
```

Output
```
Thu 2019-09-19 19:04:56.929475 [INFO ] - Guessed window DPI factor: 1
Thu 2019-09-19 19:04:56.929559 [DEBUG] - Calculated physical dimensions: 1200x800
Thu 2019-09-19 19:04:57.093585 [DEBUG] - IGNORING resize_event: 1200, 800
Thu 2019-09-19 19:04:57.106396 [DEBUG] - resize_event: 2493, 1413
Thu 2019-09-19 19:04:57.106426 [DEBUG] - not saving resolution to config because is_fullscreen is true
Viewport: Rect { x: 0.0, y: 0.0, w: 2493.0, h: 1413.0 }
Video Settings: Resolution { w: 2493.0, h: 1413.0 }
Config: (2493.0, 1413.0)
Game exited cleanly.
```

**Comment by AaronM04** [comment]
> Looks great! Two questions:
> 
>     1. Why do we not save the resolution to config when full screen?

Excellent question. If we did this, we would be losing the non-fullscreen window size the user prefers. Now, when the user turns off fullscreen, it'll be a very large window; not a good user experience.

>     2. The intro isn't centered on my system when fullscreen. Is that expected?

That's a bug. Does this happen every time? Can you give me logs?

> I ran this at the end of the `Intro` state when it transitions to `Run`. (I do agree we need to consolidate the billion different ways we store resolution.)
> 
> ```rust
> println!("Viewport: {:?}", self.viewport.get_rect());
> println!("Video Settings: {:?}", self.video_settings.get_resolution());
> println!("Config: {:?}", self.config.get_resolution());
> ```
> 
> Output
> 
> ```
> Thu 2019-09-19 19:04:56.929475 [INFO ] - Guessed window DPI factor: 1
> Thu 2019-09-19 19:04:56.929559 [DEBUG] - Calculated physical dimensions: 1200x800
> Thu 2019-09-19 19:04:57.093585 [DEBUG] - IGNORING resize_event: 1200, 800
> Thu 2019-09-19 19:04:57.106396 [DEBUG] - resize_event: 2493, 1413
> Thu 2019-09-19 19:04:57.106426 [DEBUG] - not saving resolution to config because is_fullscreen is true
> Viewport: Rect { x: 0.0, y: 0.0, w: 2493.0, h: 1413.0 }
> Video Settings: Resolution { w: 2493.0, h: 1413.0 }
> Config: (2493.0, 1413.0)
> Game exited cleanly.
> ```

**Comment by AaronM04** [comment]
@Manghi never mind about the request for logs. I can repro and am investigating! :grin:

**Comment by AaronM04** [comment]
@Manghi fixed, please test!

**Comment by Manghi** [comment]
I pushed 94cea52 to also resize the `intro_viewport`, otherwise only "CONW" was being drawn on my screen. With my giant monitor, the (stale) default resolution of 1200x800 was cutting off the "AYSTE" after re-centering once the fullscreen resolution `resize_event` triggered.

Both fullscreen and windowed were tested with this fix on my system and are working flawlessly. Please retest on your system and then feel free to merge.

**Pull request merged**